src/Microsoft.ApplicationInsights/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -40,209 +40,25 @@ public RichPayloadEventSource()
-                CopyGlobalPropertiesIfRequired(item, telemetryItem.Properties);
-                item.Sanitize();
-                this.WriteEvent(
-                    PageViewTelemetry.TelemetryName,
-                    telemetryItem.Context.InstrumentationKey,
-                    telemetryItem.Context.SanitizedTags,
-                    telemetryItem.Data,
-                    telemetryItem.Context.Flags,
-                    Keywords.PageViews);
-            }
-            else if (item is PageViewPerformanceTelemetry)
-            {
-                if (!this.EventSourceInternal.IsEnabled(EventLevel.Verbose, Keywords.PageViewPerformance))
-                {
-                    return;
-                }
-                
-                var telemetryItem = item as PageViewPerformanceTelemetry;
-                CopyGlobalPropertiesIfRequired(item, telemetryItem.Properties);
-                item.Sanitize();
-                this.WriteEvent(
-                    PageViewPerformanceTelemetry.TelemetryName,
-                    telemetryItem.Context.InstrumentationKey,
-                    telemetryItem.Context.SanitizedTags,
-                    telemetryItem.Data,
-                    telemetryItem.Context.Flags,
-                    Keywords.PageViewPerformance);
-            }
-#pragma warning disable 618
-            else if (item is SessionStateTelemetry)
-            {
-                if (!this.EventSourceInternal.IsEnabled(EventLevel.Verbose, Keywords.Events))
-                {
-                    return;
-                }
-                
-                var telemetryItem = (item as SessionStateTelemetry).Data;
-                CopyGlobalPropertiesIfRequired(item, telemetryItem.Properties);
-                item.Sanitize();
-                this.WriteEvent(
-                    EventTelemetry.TelemetryName,
-                    telemetryItem.Context.InstrumentationKey,
-                    telemetryItem.Context.SanitizedTags,
-                    telemetryItem.Data,
-                    telemetryItem.Context.Flags,
-                    Keywords.Events);
-            }
-            else if (item is AvailabilityTelemetry)
-            {
-                if (!this.EventSourceInternal.IsEnabled(EventLevel.Verbose, Keywords.Availability))
-                {
-                    return;
-                }
-                
-                var telemetryItem = item as AvailabilityTelemetry;
-                CopyGlobalPropertiesIfRequired(item, telemetryItem.Properties);
-                item.Sanitize();
-                this.WriteEvent(
-                    AvailabilityTelemetry.TelemetryName,
-                    telemetryItem.Context.InstrumentationKey,
-                    telemetryItem.Context.SanitizedTags,
-                    telemetryItem.Data,
-                    telemetryItem.Context.Flags,
-                    Keywords.Availability);
+                    eventName: RequestTelemetry.TelemetryName,
```

Comment:
This should be `eventName: richPayloadTelemetryItem.TelemetryName`

Reply:
thank you